From 5439eeffad823091bc2bd9b8ce7fa802120c7130 Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Wed, 27 Aug 2025 12:20:18 -0400
Subject: [PATCH 1/3] Revert "splice: introduce FMODE_SPLICE_READ and
 FMODE_SPLICE_WRITE" The reverted commit causes splice erroneously to fail in
 some cases. For example, when incident gets called by CTS to pipe data from
 incidentd, default_file_splice_write fails with EINVAL when it shouldn't.

The equivalent upstream commit was never merged upstream, neither in
Linux nor AOSP.

This reverts commit 21db8bb.

Issue: FP2P-408
Change-Id: I33f67b0616ce50f4fb0b96cba6352874dfa17567
---
 fs/open.c          | 4 ----
 fs/splice.c        | 6 ------
 include/linux/fs.h | 5 -----
 3 files changed, 15 deletions(-)

diff --git a/fs/open.c b/fs/open.c
index 2575d99eb0d3..a0eef4dced7b 100644
--- a/fs/open.c
+++ b/fs/open.c
@@ -693,10 +693,6 @@ static int do_dentry_open(struct file *f,
 		return 0;
 	}
 
-	if (S_ISREG(inode->i_mode))
-		f->f_mode |= FMODE_SPLICE_WRITE | FMODE_SPLICE_READ;
-
-
 	f->f_op = fops_get(inode->i_fop);
 
 	error = security_file_open(f, cred);
diff --git a/fs/splice.c b/fs/splice.c
index 9fbb58c32d04..8c847a3c3643 100644
--- a/fs/splice.c
+++ b/fs/splice.c
@@ -632,9 +632,6 @@ ssize_t default_file_splice_read(struct file *in, loff_t *ppos,
 		.spd_release = spd_release_page,
 	};
 
-	if (unlikely(!(in->f_mode & FMODE_SPLICE_READ)))
-		return -EINVAL;
-
 
 	if (splice_grow_spd(pipe, &spd))
 		return -ENOMEM;
@@ -1093,9 +1090,6 @@ static ssize_t default_file_splice_write(struct pipe_inode_info *pipe,
 {
 	ssize_t ret;
 
-	if (unlikely(!(out->f_mode & FMODE_SPLICE_WRITE)))
-		return -EINVAL;
-
 	ret = splice_from_pipe(pipe, out, ppos, len, flags, write_pipe_buf);
 	if (ret > 0)
 		*ppos += ret;
diff --git a/include/linux/fs.h b/include/linux/fs.h
index d3c4536ebac0..0a285e87e70b 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -133,11 +133,6 @@ typedef void (dio_iodone_t)(struct kiocb *iocb, loff_t offset,
 /* File was opened by fanotify and shouldn't generate fanotify events */
 #define FMODE_NONOTIFY		((__force fmode_t)0x1000000)
 
-/* File can be read using splice */
-#define FMODE_SPLICE_READ       ((__force fmode_t)0x8000000)
-/* File can be written using splice */
-#define FMODE_SPLICE_WRITE      ((__force fmode_t)0x10000000)
-
 /*
  * Flag for rw_copy_check_uvector and compat_rw_copy_check_uvector
  * that indicates that they should check the contents of the iovec are
-- 
2.45.2

